testutil/integration/network/network.go

```diff
@@ -0,0 +1,218 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+package network
+
+import (
+	"encoding/json"
+	"math"
+
+	"github.com/evmos/evmos/v14/app"
+	"github.com/evmos/evmos/v14/types"
+
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	evmtypes "github.com/evmos/evmos/v14/x/evm/types"
+	feemarkettypes "github.com/evmos/evmos/v14/x/feemarket/types"
+	infltypes "github.com/evmos/evmos/v14/x/inflation/types"
+	revtypes "github.com/evmos/evmos/v14/x/revenue/v1/types"
+)
+
+// Network is the interface that wraps the methods to interact with integration test network.
+//
+// It was designed to avoid users to access module's keepers directly and force integration tests
+// to be closer to the real user's behavior.
+type Network interface {
+	GetContext() sdktypes.Context
+	GetChainID() string
+	GetDenom() string
+	GetValidators() []stakingtypes.Validator
+
+	CommitBlock() error
+
+	GetEvmClient() evmtypes.QueryClient
+	GetRevenueClient() revtypes.QueryClient
+	GetInflationClient() infltypes.QueryClient
+	GetBankClient() banktypes.QueryClient
+	GetFeeMarketClient() feemarkettypes.QueryClient
+	GetAuthClient() authtypes.QueryClient
+
+	// Because to update the module params on a conventional manner governance
+	// would be require, we should provide an easier way to update the params
+	UpdateRevenueParams(params revtypes.Params) error
+	UpdateInflationParams(params infltypes.Params) error
+	UpdateEvmParams(params evmtypes.Params) error
+
+	BroadcastTxSync(txBytes []byte) (abcitypes.ResponseDeliverTx, error)
+	Simulate(txBytes []byte) (*txtypes.SimulateResponse, error)
+}
+
+var _ Network = (*IntegrationNetwork)(nil)
+
+type IntegrationNetwork struct {
+	cfg        Config
+	ctx        sdktypes.Context
+	app        *app.Evmos
+	validators []stakingtypes.Validator
+}
+
+// New configures and initializes a new integration Network instance with
+// the given configuration options. If no configuration options are provided
+// it uses the default configuration.
+//
+// It panics if an error occurs.
+func New(opts ...ConfigOption) *IntegrationNetwork {
+	cfg := DefaultConfig()
+	// Modify the default config with the given options
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	ctx := sdktypes.Context{}
+	network := &IntegrationNetwork{
+		cfg:        cfg,
+		ctx:        ctx,
+		validators: []stakingtypes.Validator{},
+	}
+
+	err := network.configureAndInitChain()
+	if err != nil {
+		panic(err)
```

Comment:
because if this does not work nothing else should execute, if an error is returned it could be mishandled by the user. Wanted to avoid unexpected behavior